dot/rpc/modules/system_test.go

```diff
@@ -1,4 +1,702 @@
+			exp: []interface{}{"Authority"},
+		},
+		{
+			name:   "UnknownRole",
+			fields: fields{mockNetworkAPI4, nil, nil, nil, nil, nil},
+			args: args{
+				req: &EmptyRequest{},
+			},
+			exp: []interface{}{"UnknownRole", []interface{}{uint8(21)}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var res []interface{}
+			tt.args.res = &res
+			sm := &SystemModule{
+				networkAPI: tt.fields.networkAPI,
+				systemAPI:  tt.fields.systemAPI,
+				coreAPI:    tt.fields.coreAPI,
+				storageAPI: tt.fields.storageAPI,
+				txStateAPI: tt.fields.txStateAPI,
+				blockAPI:   tt.fields.blockAPI,
+			}
+			err := sm.NodeRoles(tt.args.r, tt.args.req, tt.args.res)
+			if tt.wantErr {
+				if assert.NotNil(t, err) {
+					assert.EqualError(t, err, tt.err.Error())
+				}
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.exp, *tt.args.res)
+			}
+		})
+	}
+}
+
+func TestSystemModule_AccountNextIndex(t *testing.T) {
+	storageKeyHex := common.MustHexToBytes("0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da93116aec311d8421cece41129ffaac05aa7f9580382edb384b1b43cbcf3d1b1e7f1a1d232cf4139bd48eaafb9656da27d")
+	signedExt := common.MustHexToBytes("0xad018400d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d0146d0050619728683af4e9659bf202aeb2b8b13b48a875adb663f449f1a71453903546f3252193964185eb91c482cf95caf327db407d57ebda95046b5ef890187001000000108abcd")
+	v := make([]*transaction.ValidTransaction, 1)
+	v[0] = &transaction.ValidTransaction{
+		Extrinsic: types.NewExtrinsic(signedExt),
+		Validity:  new(transaction.Validity),
+	}
+
+	mockTxStateAPI := new(apimocks.TransactionStateAPI)
+	mockTxStateAPI.On("Pending").Return(v, nil)
+
+	mockCoreAPI := new(apimocks.CoreAPI)
+	mockCoreAPI.On("GetMetadata", (*common.Hash)(nil)).Return(common.MustHexToBytes(testdata.GetTestData()), nil)
+
+	mockCoreAPIErr := new(apimocks.CoreAPI)
+	mockCoreAPIErr.On("GetMetadata", (*common.Hash)(nil)).Return(nil, errors.New("getMetadata error"))
+
+	// Magic number mismatch
+	mockCoreAPIMagicNumMismatch := new(apimocks.CoreAPI)
+	mockCoreAPIMagicNumMismatch.On("GetMetadata", (*common.Hash)(nil)).Return(storageKeyHex, nil)
+
+	mockStorageAPI := new(apimocks.StorageAPI)
+	mockStorageAPI.On("GetStorage", (*common.Hash)(nil), storageKeyHex).Return(common.MustHexToBytes("0x03000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"), nil)
+
+	mockStorageAPIErr := new(apimocks.StorageAPI)
+	mockStorageAPIErr.On("GetStorage", (*common.Hash)(nil), storageKeyHex).Return(nil, errors.New("getStorage error"))
+
+	type fields struct {
+		networkAPI NetworkAPI
+		systemAPI  SystemAPI
+		coreAPI    CoreAPI
+		storageAPI StorageAPI
+		txStateAPI TransactionStateAPI
+		blockAPI   BlockAPI
+	}
+	type args struct {
+		r   *http.Request
+		req *StringRequest
+		res *U64Response
+	}
+	tests := []struct {
+		name    string
+		fields  fields
```

Comment:
you can replace fields with just `SystemModule` and use the `newSystemModule` constructor.